pkg/agent/plugin/nodeattestor/aws/iid.go

```diff
@@ -60,53 +60,56 @@ func httpGetBytes(url string) ([]byte, error) {
 	return bytes, nil
 }
 
-func (p *IIDAttestorPlugin) FetchAttestationData(ctx context.Context, req *nodeattestor.FetchAttestationDataRequest) (*nodeattestor.FetchAttestationDataResponse, error) {
+func (p *IIDAttestorPlugin) FetchAttestationData(stream nodeattestor.NodeAttestor_FetchAttestationData_PluginStream) error {
 	p.mtx.RLock()
 	defer p.mtx.RUnlock()
 
+	_, err := stream.Recv()
```

Comment:
It feels a little weird to me that we have to read a message off the stream before doing anything. As a plugin developer, I would only expect a Recv to be required if I know a challenge is coming...

Reply:
good point. lemme give it a think.